runtime/bcutil/defineclass.c

```diff
@@ -928,7 +930,11 @@ createROMClassFromClassFile(J9VMThread *currentThread, J9LoadROMClassData *loadD
 			J9_VM_FUNCTION(currentThread, setCurrentException)(currentThread, exceptionNumber, NULL);
 		} else {
 			PORT_ACCESS_FROM_JAVAVM(vm);
-			J9_VM_FUNCTION(currentThread, setCurrentExceptionUTF)(currentThread, exceptionNumber, (const char*)errorUTF);
+			if (result == BCT_ERR_DUPLICATE_NAME) {
+				J9_VM_FUNCTION(currentThread, setCurrentExceptionNLSWithArgs)(currentThread, J9NLS_JCL_DUPLICATE_CLASS_DEFINITION, exceptionNumber, (const char*)errorUTF, strlen(errorUTF));
```

Comment:
Since you are checking that its duplicate name here you could just place the correct excpetion type instead of `exceptionNumber`